src/ucan/lookup.ts

```diff
@@ -0,0 +1,78 @@
+import * as Path from "../path/index.js"
+import { Dictionary, Ucan } from "./types.js"
+
+////////
+// 🛠️ //
+////////
+
+export function fsWriteUcans(collection: Ucan[]): Ucan[] {
+  return collection.filter(ucan => ucan.payload.att.some(cap => cap.with.scheme === "wnfs"))
+}
+
+export function lookupFsWriteUcan(
+  collection: Ucan[],
+  fileSystemDID: string,
+  path: Path.DistinctivePath<Path.Segments>
+): Ucan | null {
+  return lookupFsUcan(
+    fsWriteUcans(collection),
+    pathSoFar => ucan => {
+      const hierPart = `//${fileSystemDID}/${Path.toPosix(pathSoFar)}`
+
+      return !!ucan.payload.att.find(cap => {
+        return cap.with.hierPart === hierPart && (cap.can === "*" || cap.can.namespace === "fs")
+      })
+    },
+    fileSystemDID,
+    path
+  )
+}
+
+export function rootIssuer(ucan: Ucan, ucanDictionary: Dictionary): string {
+  if (ucan.payload.prf.length) {
+    return ucan.payload.prf.reduce(
+      (acc, prf) => {
+        // Always prefer the first proof.
+        // TBH, not sure what's best here.
+        if (acc) return acc
+
+        const prfUcan = ucanDictionary[prf]
+        if (!prfUcan) throw new Error("Missing a UCAN in the repository")
+
+        return rootIssuer(prfUcan, ucanDictionary)
+      }
+    )
```

Comment:
Yeah this function is lacking in several areas. Ideally it should use the implementation in the used UCAN library. It's just to get the root issuer. I don't see a use case in our ecosystem, but since the SDK can be used with other ecosystems as well, I'm just preparing for any possibility.

Reply:
Makes sense. If I were to guess, other ecosystems would want to (optionally) pass in a capability as a discriminant to do determine root issuer for what? If they all have the same capability, it probably doesn't matter.

Reply:
Yeah maybe. I also made sure that this function is only used in the implementations themselves. So if people were to write another account system implementation, they can use something else.

Reply:
Got it, yeah. That sounds more flexible for folks in general. 👌